### PR TITLE
Image Grid: Add measurements, fallback support and correct default spacing mismatch

### LINKED ISSUE #176: Image Grid, Simple Masonry: Add ability to override default alt text for image
You should be able to override the default alt text for an image so you can use more contextually relevant alt text.

### PR DESCRIPTION
**Pull request by AlexGStapleton**
This PR resolves #176, #193, #189.

#193 will require a follow up premium PR for lightbox - tooltips already work.

### CODE REVIEW

File: widgets/image-grid/tpl/default.php [hunk 1]
```diff
@@ -20,9 +19,7 @@
 						<?php endif; ?>
 					<?php endforeach; ?>>
 				<?php endif; ?>
-				<?php echo wp_get_attachment_image( $image['image'], $attachment_size, false, array(
-					'title' => $image['title']
-				) );?>
+				<?php echo $image['image_html'];	?>
```
Comment (Misplon) [inline widgets/image-grid/tpl/default.php]: @AlexGStapleton Can you check out the additional spaces before the PHP close when able. Thanks mate.

File: widgets/image-grid/image-grid.php [hunk 2]
```diff
@@ -80,29 +79,29 @@ function get_widget_form(){
...
+						'label' => __( 'Image size', 'so-widgets-bundle' ),
 						'type' => 'image-size',
 						'default' => 'full',
 					),
 
 					'max_height' => array(
-						'label' => __('Maximum image height', 'so-widgets-bundle'),
-						'type' => 'number',
+						'label' => __( 'Maximum image height', 'so-widgets-bundle' ),
+						'type' => 'measurement',
```
[the remark this answers is not in the document]
Comment (AlexGStapleton) [reply widgets/image-grid/image-grid.php]: I had a look around and I couldn't seem to find any migration code. 😕

[the remark this answers is not in the document]
Comment (braamgenis) [reply widgets/image-grid/image-grid.php]: You'd override the [`modify_instance`](https://github.com/siteorigin/so-widgets-bundle/blob/develop/base/siteorigin-widget.class.php#L1206) function and check whether there's an existing value for `max_height`, and if there is, and it doesn't already have a unit, append the default unit. Remember '0' and 0 are considered empty values, but they're valid values.

**Comment by braamgenis** [comment]
Ok, so the code looks good, but I'm concerned that the change for #189 is going to affect the spacing for existing grid layouts. It's either going to cause a lot of support tickets and/or upset people when they update and it changes their layout.

**Comment by AlexGStapleton** [comment]
To be honest, I don't think we'll more than a few emails.
@Misplon Thoughts? There's a setting to control the spacing so we could write a canned response if needed or if you would rather not risk it, I can revert it.

**Comment by gregpriday** [comment]
I'm happy with a small change in existing image grids if this gives a more expected behaviour.

**Pull request merged**